### PR TITLE
Run unit tests on AppVeyor and Travis.

### PR DESCRIPTION
**Pull request by PJB3005**
Also Travis now runs on OSX too.

Also thanks to @psykzz for helping me get logs in Python to work.

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -12,4 +17,8 @@ build:
   parallel: true
   verbosity: minimal
 
-test: off
+#after_build:
+#- cmd: py -3.5 Resources\buildResourcePack.py --resources-dir .\Resources --out .\Resources\ResourcePack.zip --sprite-renderer .\Resources\SpriteRenderer\MSpriteRenderer.exe --atlas-tool .\Tools\AtlasTool.exe --to-stderr
```
Comment (psykzz) [inline .appveyor.yml:21]: This meant to be disabled?

**Comment by ZoldorfTheWizard** [comment]
Very good work all

File: .appveyor.yml (hunk 1 above)
Comment (PJB3005) [reply .appveyor.yml:21]: It doesn't work anyways. I guess I could disable animation generation but that won't get us much will it.

**Comment by PJB3005** [comment]
Alright we're good here.

File: .travis.yml [hunk 2]
```diff
@@ -1,2 +1,21 @@
...
+addons:
+  apt:
+    packages:
+    - nunit-console
+
+env:
+- HEADLESS=1
+
+before_script:
+- "curl http://84.195.252.227/static/ResourcePack.zip -o Resources/ResourcePack.zip"
```
Comment (ComicIronic) [inline .travis.yml:16]: Why the direct IP and why this?

Comment (PJB3005) [reply .travis.yml:16]: Well i had to host the file somewhere and hosting it in the repo feels wrong. If you have a better idea tell me.

**Review by ZoldorfTheWizard — APPROVED** [review]

File: .travis.yml (hunk 2 above)
Comment (ComicIronic) [reply .travis.yml:16]: Isn't the ResourcePack made out of stuff in the repo in the first place? Can't you generate it when you need it?

Comment (ZoldorfTheWizard) [reply .travis.yml:16]: the sprite renderer does not fully work on linux and macos, so things like animated sprites dont get rendered into the texture atlasses. He downloads a precompiled pack to work around it for now

**Pull request merged**